Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Update/Internal/FbUpdateSqlGenerator.cs

```diff
@@ -0,0 +1,364 @@
+           int commandPosition)
+        {
+            Check.NotNull(commandStringBuilder, nameof(commandStringBuilder));
+            Check.NotEmpty(modificationCommands, nameof(modificationCommands));
+            var name = modificationCommands[0].TableName;
+            var schema = modificationCommands[0].Schema;
+
+            commandStringBuilder.AppendLine("EXECUTE BLOCK RETURNS (AffectedRows INT) AS BEGIN")
+                                .AppendLine($"AffectedRows=0;");
+
+            for (var i = 0; i < modificationCommands.Count; i++)
+            {
+                var operations = modificationCommands[i].ColumnModifications;
+                var writeOperations = operations.Where(o => o.IsWrite).ToArray();
+                var readOperations = operations.Where(o => o.IsRead).ToArray();
+                commandStringBuilder.Append($"DELETE FROM {SqlGenerationHelper.DelimitIdentifier(name)} ");
+                commandStringBuilder.AppendLine($" WHERE {SqlGenerationHelper.DelimitIdentifier(operations.First().ColumnName)}={operations[0].Value}; ");
+                AppendUpdateOutputClause(commandStringBuilder);
+            }
+            commandStringBuilder.AppendLine("SUSPEND;");
+            commandStringBuilder.AppendLine("END;");
+            return ResultSetMapping.NotLastInResultSet;
+        } 
+
+        private void AppendUpdateOutputClause(StringBuilder commandStringBuilder)
+        {
+            //Increment of updates 
+            commandStringBuilder
+                    .AppendLine("IF (ROW_COUNT > 0) THEN")
+                    .AppendLine("   AffectedRows=AffectedRows+ROW_COUNT;");
+
+        } 
+
+        private void AppendInsertOutputClause(
+            StringBuilder commandStringBuilder,
+            string name,
+            string schema,
+            IReadOnlyList<ColumnModification> operations,
+            IReadOnlyList<ColumnModification> allOperations)
+        {
+            if (allOperations.Count > 0 && allOperations[0] == operations[0])
+            {
+                commandStringBuilder
+                    .AppendLine($" RETURNING {SqlGenerationHelper.DelimitIdentifier(operations.First().ColumnName)} INTO :AffectedRows;")
+                    .AppendLine("IF (ROW_COUNT > 0) THEN")
+                    .AppendLine("   SUSPEND;");
+            }
+        }
+
+        protected override ResultSetMapping AppendSelectAffectedCountCommand(StringBuilder commandStringBuilder, string name,
+            string schema, int commandPosition)
+        {
+            // Not Implemented!
+            return ResultSetMapping.LastInResultSet;
+        }
+
+        public override void AppendBatchHeader(StringBuilder commandStringBuilder)
+        {
+            //Insert FbSQL Fast(Insert/Update) 
+        }
+
+        protected override void AppendIdentityWhereCondition(StringBuilder commandStringBuilder, ColumnModification columnModification)
+        {
+            // Not Implemented!
```

Comment:
Why not to throw an exception?

Reply:
Yes we can. Or simply remove. I've done a lot of improvement in this class.
As soon as Jean completes some of his suggestions I'll ask him to get this from my fork.
And then we were in sync with the officer.